stays/views.py

```diff
@@ -78,3 +82,44 @@ def get(self, request, stay_id):
             "check_out"  : room[0].check_out
         }
         return JsonResponse(data, status=200)
+
+class StayRoomsView(View):
+    def get(self, request, stay_id):
+        if not Staytype.objects.filter(id = stay_id).exists():
+            return JsonResponse({"message":"INVALID_ID"}, status=404)
+
+        check_in , check_out = request.GET.get('CheckIn'),request.GET.get('CheckOut')
+
+        if not (check_in and check_out):
+            return JsonResponse({"message":"INVALID_DATE"}, status=400)
+
+        check_in  = datetime.strptime(check_in,"%Y-%m-%d")
+        check_out = datetime.strptime(check_out,"%Y-%m-%d")
+
+        if check_in > check_out:
+            return JsonResponse({"message":"INVALID_DATE"}, status=400)
+
+        q1 = Q(book__check_in__gte=check_in) & Q(book__check_in__lte=check_out)
+        q2 = Q(book__check_out__gte=check_in) & Q(book__check_out__lte=check_out)
+        q3 = Q(book__check_in__lt=check_in,book__check_out__gt=check_out)
+
+        rooms = Room.objects.annotate(
+            remain       = F('quantity')-Count('book__id',filter=(q1|q2|q3)),
+            is_available = Case(
+                When(
+                    remain = 0,
+                    then   = False), default = True)).filter(staytype__id=stay_id).exclude(is_available=False)
```

Comment:
1. `filter(staytype__id)`를 앞으로 옮겨주세요. 호출되는 쿼리는 다르지 않겠지만 코드를 읽는 측면에서 `statype_id=1`인 room 중에서 `remain`, `is_available` column을 가져오는게 읽기 편할 것 같습니다.
2. `filter(staytype__id=stay_id)` ==> filter(staytype_id=stay_id)로 수정하는게 좋을 것 같습니다. 전자는 room에서 staytype 테이블에 있는 id를 가져오는 경우고, 후자는 room 테이블 안에 `staytype_id` colume을 기준으로 가져오는 경우입니다.
```suggestion
        rooms = Room.objects.filter(staytype_id=stay_id).annotate(
            remain       = F('quantity')-Count('book__id',filter=(q1|q2|q3)),
            is_available = Case(
                When(
                    remain = 0,
                    then   = False), default = True)).exclude(is_available=False)
```